main.tf

```diff
@@ -39,10 +39,12 @@ resource "aws_route53_zone" "zone" {
 locals {
   records_expanded = [
     for record in var.records : merge({
-      record_id = join("-", compact([lower(record.type), lower(record.name)]))
-      ttl       = null
-      records   = null
-      alias     = null
+      record_id      = join("-", compact([lower(record.type), lower(record.name), try("${lower(record.set_identifier)}-${lower(record.weight)}", null)]))
```

Comment:
Not sure yet if this is really the right approach to go. I will decide later once I applied all the different policies.